src/shared/udev-util.c

```diff
@@ -119,10 +120,18 @@ static int device_monitor_handler(sd_device_monitor *monitor, sd_device *device,
 
         assert(device);
         assert(data);
-        assert(data->sysname);
+        assert(data->sysname || (data->devlink));
         assert(!data->device);
 
-        if (sd_device_get_sysname(device, &sysname) >= 0 && streq(sysname, data->sysname)) {
+        if (data->devlink) {
+                const char *devlink;
+
+                FOREACH_DEVICE_DEVLINK(device, devlink)
```

Comment:
i think we should support loking for both symlinks and the main device node here, given that effectively both paths can be used interchangably in most uses. i.e. sd_device_get_devname() should satisfy the path we look for too